core/dictgen/src/rootcling_impl.cxx

```diff
@@ -3664,6 +3600,20 @@ class TRootClingCallbacks : public cling::InterpreterCallbacks {
          }
       }
    }
+
+   virtual void EnteredSubmodule(clang::Module* M,
```

Comment:
Could you add some doc on why this needs to be done?

Could `rootcling` just unconditionally make `Core` visible? (That seems to be in line with its non-modular behavior.)

Reply:
Done.

When rootcling starts building a module, it becomes resilient to the outside environment and pre-included files have no effect. This hook informs rootcling when a new submodule is being built so that it can make Core.Rtypes.h visible.                                               |